src/main/java/frc/robot/commands/AlgaeCommand.java

```diff
@@ -1,45 +1,49 @@
-// Copyright (c) FIRST and other WPILib contributors.
-// Open Source Software; you can modify and/or share it under the terms of
-// the WPILib BSD license file in the root directory of this project.
-
 package frc.robot.commands;
 
-import java.util.function.Supplier;
-
 import edu.wpi.first.wpilibj2.command.Command;
 import frc.robot.subsystems.AlgaeSubsystem;
 
-/* You should consider using the more terse Command factories API instead https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#defining-commands */
+
 public class AlgaeCommand extends Command {
-  AlgaeSubsystem algae;
-  Supplier<Double> percent;
-  /** Creates a new AlgaeCommand. */
-  public AlgaeCommand(Supplier<Double> percent) {
-    this.percent = percent;
-    algae = AlgaeSubsystem.getInstance();
+
+  private final AlgaeSubsystem algaeManipulator;
+  private double angle;
+  private double ratio;
+
+  public AlgaeCommand(double angle, double gearRatio) {
```

Comment:
You can have a separate different Algae command and keep the current AlgaeCommand, since your command has different constructor from current one. 
Then we can have different button mapping to test both of them to decide which one is better.